tensorflow_lite_support/c/task/vision/image_classifier.cc

```diff
@@ -0,0 +1,223 @@
+  } else {
+    return nullptr;
+  }
+}
+
+TfLiteImageClassifier* TfLiteImageClassifierFromFile(const char* model_path) {
+  TfLiteImageClassifierOptions* default_options =
+      TfLiteImageClassifierOptionsDefault();
+  TfLiteImageClassifierOptionsSetModelFilePath(default_options, model_path);
+  return TfLiteImageClassifierFromOptions(default_options);
+}
+
+TfLiteClassificationResult* GetClassificationResultCStruct(
+    const ClassificationResultCpp classification_result_cpp) {
+  auto* c_classifications =
+      new TfLiteClassifications[classification_result_cpp
+                                    .classifications_size()];
+
+  for (int head = 0; head < classification_result_cpp.classifications_size();
+       ++head) {
+    const ClassificationsCpp& classifications =
+        classification_result_cpp.classifications(head);
+    c_classifications[head].head_index = head;
+
+    auto* c_categories = new TfLiteCategory[classifications.classes_size()];
+    c_classifications->size = classifications.classes_size();
+
+    for (int rank = 0; rank < classifications.classes_size(); ++rank) {
+      const ClassCpp& classification = classifications.classes(rank);
+      c_categories[rank].index = classification.index();
+      c_categories[rank].score = classification.score();
+
+      if (classification.has_class_name())
+        c_categories[rank].label = strdup(classification.class_name().c_str());
+      else
+        c_categories[rank].label = nullptr;
+
+      if (classification.has_display_name())
+        c_categories[rank].display_name =
+            strdup(classification.display_name().c_str());
+      else
+        c_categories[rank].display_name = nullptr;
+    }
+    c_classifications[head].categories = c_categories;
+  }
+
+  TfLiteClassificationResult* c_classification_result =
+      new TfLiteClassificationResult;
+  c_classification_result->classifications = c_classifications;
+  c_classification_result->size =
+      classification_result_cpp.classifications_size();
+
+  return c_classification_result;
+}
+
+TfLiteClassificationResult* TfLiteImageClassifierClassify(
+    const TfLiteImageClassifier* classifier,
+    const TfLiteFrameBuffer* frame_buffer) {
+  std::unique_ptr<FrameBufferCpp> cpp_frame_buffer =
+      CreateCppFrameBuffer(frame_buffer);
+
+  if (cpp_frame_buffer == nullptr) return nullptr;
+
+  StatusOr<ClassificationResultCpp> classification_result_cpp =
+      classifier->impl->Classify(*cpp_frame_buffer);
+
+  if (!classification_result_cpp.ok()) return nullptr;
+
+  TfLiteClassificationResult* c_classification_result =
+      GetClassificationResultCStruct(
+          ClassificationResultCpp(classification_result_cpp.value()));
+
+  return c_classification_result;
+}
+
+TfLiteClassificationResult* TfLiteImageClassifierClassifyWithRoi(
+    const TfLiteImageClassifier* classifier,
+    const TfLiteFrameBuffer* frame_buffer, const TfLiteBoundingBox* roi) {
```

Comment:
These function input params cannot be null. So either check if they're null, or change pointer to reference.

Same for all other functions.

Reply:
AFAIK, C does not have pass by reference. I will change the private methods used by the C++ code. Not the ones that are declared in C header files. Also, I thought null check for function params would be overkill. Since there is a chance that the programmer can pass an invalid address and still get a seg fault. A null check won't prevent that. Followed the pattern of params in [nl_classifier](https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/c/task/text/nl_classifier.cc#L69-L72)

I'll still add the null check for params when I push the fixes for these comments and we can keep it that way for vision. But since it won't prevent a seg fault from happening because of invalid addresses, we may consider adding a note about that in the documentation.
C APIs are meant to be used with wrappers. So the onus can be on the caller framework to not pass an explicit null or an invalid address.

Reply:
Added null check for params. Changed all methods to pass by reference sparing the public C APIs.

Reply:
Yeah, you're right. Reference is not supported  in C.

The main concern for null pointer is that, Task library can return null ImageClassifier and FrameBuffer, therefore, users may just pass whatever they get to other methods of Task Library. In the future, we should also implementing  error reporting, so that the error message can be properly surfaced.